File: openadmin/spec/section.py
# SPDX-FileCopyrightText: 2026 OpenAdmin
#
# SPDX-License-Identifier: AGPL-3.0-or-later


from typing import TypedDict

from .icons import Icon
from .page import Page


class Section(TypedDict):
    id: str
    name: str
    description: str | None
    icon: Icon | None
    pages: list[Page]

File: client/src/schemas/section.ts
// SPDX-FileCopyrightText: 2026 OpenAdmin
//
// SPDX-License-Identifier: AGPL-3.0-or-later

import { z } from "zod"
import { iconSchema } from "./icon"
import { pageSchema } from "./page"

export const sectionSchema = z.object({
	id: z.string(),
	name: z.string(),
	description: z.string().nullable(),
	icon: iconSchema.nullable(),
	pages: z.array(pageSchema),
})

export type Section = z.infer<typeof sectionSchema>
